### DISCUSSION TITLE
Question: is it possible to have a screen with the dual function `create` and `edit`?

### DISCUSSION THREAD

**Discussion by gvlx** [Budibase/budibase#3845] [Help] [answered]
Hi,

Most of my screens deal with creating or editing a form.

But those screens must be developed in pairs, as practically all elements are the same.

What changes in the `edit` screen, compared to the `create` screen:
- the form does have data to fill in the fields;
- the form type changes from `Create` to `Update` (can be changed in `Conditions`)
- the save action must ensure all data is saved, not only the shown fields.

What are you suggestions to create an fully operational `create/edit` screen?
- `:id` url parameter set to "new" on create forms?
- more?

**Comment by cah-keith-knight** [comment]
I'd love to know the answer to this too! =)

**Comment by melohagan** [comment, marked as answer]
Hey @cah-keith-knight 

[Table blocks](https://docs.budibase.com/docs/table-block) were built for this, as they allow you to add 'New' and 'Edit' [Side panels](https://docs.budibase.com/docs/side-panel)

This allows you to keep CRUD functionality in one screen!
